libs/libapi/src/libapi/utils.py

```diff
@@ -155,16 +160,80 @@ def to_rows_list(
             cached_assets_base_url=cached_assets_base_url,
             cached_assets_directory=cached_assets_directory,
             offset=offset,
+            row_idx_column=row_idx_column,
         )
     except Exception as err:
         raise TransformRowsProcessingError(
             "Server error while post-processing the split rows. Please report the issue."
         ) from err
     return [
         {
-            "row_idx": idx + offset,
+            "row_idx": idx + offset if row_idx_column is None else row.pop(row_idx_column),  # type: ignore
```

Comment:
Actually, `/rows` doesn't need `row_idx_column` (it depends on the pagination offset and iteration).
I put `# type: ignore` because of mypy warning about `Mapping` without `pop` method, not sure if I could cast it to dict, I would like to perform pop in order to take advantage and remove the column to use row without it in the next line.

Reply:
OK! maybe you could change the type of Row (Dict instead of Mapping): https://github.com/huggingface/datasets-server/blob/694b00e4bb5b4c3cee6d8f5706eb148d0e77f7f7/libs/libcommon/src/libcommon/utils.py#L85

Reply:
Done in https://github.com/huggingface/datasets-server/pull/1516/commits/d4babea14050b03679037cb9ae35bb7e602dc2e8